### ISSUE TITLE
buffer v5.x dependency crashes on Android 8 and earlier

### ISSUE THREAD

**Issue by noahfpf** [rollbar/rollbar-react-native#81]
In both the emulator and on devices running older versions of Android (tested with 8.0 and 6.0), our application crashes when importing from `buffer` (5.4.2), which is a dependency of `rollbar-react-native` (version 0.5.0).

The crash comes at this line:
```
Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype)
```

https://github.com/feross/buffer/blob/v5.4.2/index.js#L186

And just before that, the following warning is logged by `buffer`:
> "This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support."

I think we've fixed this problem by requiring buffer `^4.9.1` instead, though I'm not yet sure what repercussions that might have.  We added this to `package.json`:

```
  "resolutions": {
    "rollbar-react-native/buffer": "^4.9.1",
  },
```

Possibly related: https://github.com/facebook/react-native/issues/1424

Any idea whether buffer v4 is sufficient for basic rollbar logging and exception catching?

Relevant log lines:
```
09-09 14:58:33.897 27501 27558 E ReactNativeJS: This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.
09-09 14:58:33.901 27501 27558 E ReactNativeJS: undefined is not a function (evaluating 'Object.setPrototypeOf(s.prototype,Uint8Array.prototype)')
```

**Comment by waltjones** [comment]
@noahfpf Thanks for the report. Seems strange, since that dependency was added in Sep. 2017 when Android 8 (and earlier) were pretty mainstream.

I'll test here and see if 4.x is sufficient.

**Comment by spacesuitdiver** [comment]
Worth noting this version pinning only works with yarn it seems.

**Comment by waltjones** [comment]
Fixed in 0.6.1. PR: https://github.com/rollbar/rollbar-react-native/pull/83
